### PR TITLE
[Google Blockly] CT-435: Replace all instances of block-id with data-id in Minecraft to fix callouts

### PR DESCRIPTION
**Pull request by ebeastlake**
When we flipped Minecraft, we forgot to update the callouts' selectors so they would render correctly.

@molly-moen used a script to do this for Sprite Lab since there were so many levels that needed to be updated: https://github.com/code-dot-org/code-dot-org/pull/54642 

Since only the first Minecraft tutorial uses callouts (Adventurer), we manually updated these levels. I tested them using the `?show_callouts=1` parameter on each level. 

There is unnecessary diff in the level files since I made the edits through levelbuilder. For each file, the two relevant changes are:
1. Replacing block-id with data-id on the callout
2. Adding id=[data-id] on the block to indicate where the callout should render

Note: Adding the `id` on the `when run` block appeared to render the callout in pretty much every case correctly. There is one level where the callout is more over the workspace in Google Blockly vs. closer to the toolbox in CDO, but since the callout is general, I don't think it matters. 

## Links

Jira ticket: https://codedotorg.atlassian.net/browse/CT-435

## Testing story

These 35 files have five distinct levels, each needing to be updated in seven places (the base level and then each of 2018, 2019, 2020, 2021, 2022, and 2023). So, these screenshots represent all of the new callouts in action.

Screenshot 2024-03-28 at 4 06 48 PM

Screenshot 2024-03-28 at 4 06 31 PM

Screenshot 2024-03-28 at 4 07 08 PM

Screenshot 2024-03-28 at 4 07 21 PM

Screenshot 2024-03-28 at 4 07 59 PM

## PR Checklist:

- [x] Tests provide adequate coverage
- [x] Privacy and Security impacts have been assessed
- [x] Code is well-commented
- [x] New features are translatable or updates will not break translations
- [x] Relevant documentation has been added or updated
- [x] User impact is well-understood and desirable
- [x] Pull Request is labeled appropriately
- [x] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: dashboard/config/levels/custom/craft/Overworld Chop Trees.level [hunk 1]
```diff
@@ -62,7 +62,7 @@
   <blocks>
     <start_blocks>
       <xml>
-        <block type="when_run" deletable="false" movable="false">
+        <block type="when_run" deletable="false" movable="false" id="5">
```
Comment (ebeastlake) [inline dashboard/config/levels/custom/craft/Overworld Chop Trees.level:65]: 🆔

File: dashboard/config/levels/custom/craft/Overworld Chop Trees2022.level [hunk 2]
```diff
@@ -64,7 +69,7 @@
   <blocks>
     <start_blocks>
       <xml>
-        <block type="when_run" deletable="false" movable="false">
+        <block type="when_run" deletable="false" movable="false" id="5">
```
Comment (ebeastlake) [inline dashboard/config/levels/custom/craft/Overworld Chop Trees2022.level:72]: 🆔

File: dashboard/config/levels/custom/craft/Overworld Chop Trees_2018.level [hunk 3]
```diff
@@ -48,16 +49,23 @@
...
+  "level_concept_difficulty": {
+  }
 }]]></config>
   <blocks>
     <start_blocks>
       <xml>
-        <block type="when_run" deletable="false" movable="false">
+        <block type="when_run" deletable="false" movable="false" id="5">
```
Comment (ebeastlake) [inline dashboard/config/levels/custom/craft/Overworld Chop Trees_2018.level:68]: 🆔

File: dashboard/config/levels/custom/craft/Overworld Chop Trees_2019.level [hunk 4]
```diff
@@ -64,7 +69,7 @@
   <blocks>
     <start_blocks>
       <xml>
-        <block type="when_run" deletable="false" movable="false">
+        <block type="when_run" deletable="false" movable="false" id="5">
```
Comment (ebeastlake) [inline dashboard/config/levels/custom/craft/Overworld Chop Trees_2019.level:72]: 🆔

File: dashboard/config/levels/custom/craft/Overworld Chop Trees_2020.level [hunk 5]
```diff
@@ -64,7 +69,7 @@
   <blocks>
     <start_blocks>
       <xml>
-        <block type="when_run" deletable="false" movable="false">
+        <block type="when_run" deletable="false" movable="false" id="5">
```
Comment (ebeastlake) [inline dashboard/config/levels/custom/craft/Overworld Chop Trees_2020.level:72]: 🆔

File: dashboard/config/levels/custom/craft/Overworld Chop Trees_2021.level [hunk 6]
```diff
@@ -64,7 +69,7 @@
   <blocks>
     <start_blocks>
       <xml>
-        <block type="when_run" deletable="false" movable="false">
+        <block type="when_run" deletable="false" movable="false" id="5">
```
Comment (ebeastlake) [inline dashboard/config/levels/custom/craft/Overworld Chop Trees_2021.level:72]: 🆔

File: dashboard/config/levels/custom/craft/Overworld Chop Trees_2023.level [hunk 7]
```diff
@@ -63,7 +69,7 @@
   <blocks>
     <start_blocks>
       <xml>
-        <block type="when_run" deletable="false" movable="false">
+        <block type="when_run" deletable="false" movable="false" id="5">
```
Comment (ebeastlake) [inline dashboard/config/levels/custom/craft/Overworld Chop Trees_2023.level:72]: 🆔

File: dashboard/config/levels/custom/craft/Overworld House Frame Chosen.level [hunk 8]
```diff
@@ -71,7 +74,7 @@
   <blocks>
     <start_blocks>
       <xml>
-        <block type="when_run" deletable="false" movable="false">
+        <block type="when_run" deletable="false" movable="false" id="12">
```
Comment (ebeastlake) [inline dashboard/config/levels/custom/craft/Overworld House Frame Chosen.level:77]: 🆔

File: dashboard/config/levels/custom/craft/Overworld Chop Trees2022.level [hunk 9]
```diff
@@ -33,7 +34,7 @@
     "ideal": "15",
     "embed": "false",
     "free_play": "false",
-    "callout_json": "[\r\n {\r\n  \"localization_key\": \"craft_callout_preplaced\",\r\n  \"callout_text\": \"We've put some blocks here to get you started\",\r\n  \"element_id\": \"[block-id=\\\"5\\\"]\",\r\n  \"on\": \"\",\r\n  \"qtip_config\": {\r\n   \"style\": {\r\n    \"classes\": \"\"\r\n   },\r\n   \"position\": {\r\n    \"my\": \"left center\",\r\n    \"at\": \"right center\",\r\n    \"adjust\": {\r\n     \"x\": 0,\r\n     \"y\": 0\r\n    }\r\n   }\r\n  }\r\n }\r\n]",
```
Comment (ebeastlake) [inline dashboard/config/levels/custom/craft/Overworld Chop Trees2022.level:36]: In the files like these, here's the first change that matters.

File: dashboard/config/levels/custom/craft/Overworld Chop Trees2022.level [hunk 10]
```diff
@@ -64,7 +69,7 @@
   <blocks>
     <start_blocks>
       <xml>
-        <block type="when_run" deletable="false" movable="false">
```
Comment (ebeastlake) [inline dashboard/config/levels/custom/craft/Overworld Chop Trees2022.level:67]: And here's the second.

**Review by mikeharv — APPROVED** [review]
Looks good. I didn't check every level, but since the changes were mostly done using levelbuilder, there's a very low chance of any unexpected problems. Thank you for the thorough fix!

**Pull request merged**